test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java

```diff
@@ -0,0 +1,135 @@
+
+import java.nio.channels.Pipe;
+import java.util.*;
+
+public class PullSizeTest extends BaseConf {
+
+    private static Logger logger = Logger.getLogger(PullSizeTest.class);
+
+    private static final Map<MessageQueue, Long> OFFSE_TABLE = new HashMap<MessageQueue, Long>();
+
+    public static final String PULL_SIZE_TOPIC = "TopicPullTest";
+
+    public static final String PULL_SIZE_GROUP = "pullSizeTest";
+
+    public boolean send() throws MQClientException, InterruptedException {
+        DefaultMQProducer producer = new DefaultMQProducer(PULL_SIZE_GROUP);
+        producer.setNamesrvAddr(nsAddr);
+        producer.start();
+        int successCount = 0;
+        for (int i = 0; i < 1000; i++) {
+            try {
+                Message msg = new Message(PULL_SIZE_TOPIC, "TagA", ("RocketMQ pull size test index " + i).getBytes(RemotingHelper.DEFAULT_CHARSET));
+                SendResult sendResult = producer.send(msg);
+                if (sendResult.getSendStatus().equals(SendStatus.SEND_OK)) {
+                    successCount++;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                Thread.sleep(500);
+            }
+        }
+        producer.shutdown();
+        if (successCount > 800) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    public void pullMsg() throws MQClientException {
+        boolean result = false;
+        DefaultMQPullConsumer consumer = new DefaultMQPullConsumer(PULL_SIZE_GROUP);
+        consumer.setNamesrvAddr(nsAddr);
+        consumer.start();
+        Set<MessageQueue> mqs = consumer.fetchSubscribeMessageQueues(PULL_SIZE_TOPIC);
+        for (MessageQueue mq : mqs) {
+            if (result) {
+                break;
+            }
+            try {
+                PullResult pullResult = consumer.pull(mq, null, getMessageQueueOffset(mq), 32);
+                putMessageQueueOffset(mq, pullResult.getNextBeginOffset());
+                switch (pullResult.getPullStatus()) {
+                    case FOUND:
+                        List<MessageExt> msgFoundList = pullResult.getMsgFoundList();
+                        if (msgFoundList != null) {
+                            logger.info("[RECV] received msg queue:" + mq.getBrokerName() + "-" + mq.getQueueId());
+                            result |= msgFoundList.size() >= 32;
+                        }
+                        break;
+                    case NO_MATCHED_MSG:
+                        break;
+                    case NO_NEW_MSG:
+                        break;
+                    case OFFSET_ILLEGAL:
+                        break;
+                    default:
+                        break;
+                }
+            } catch (Exception e) {
+                logger.error("[SEND] send failed", e);
+                e.printStackTrace();
+            }
+        }
+        consumer.shutdown();
+        Assert.assertTrue(result);
```

Comment:
Could we assert the pull size is what you want, while not the simple true or false ?